### DISCUSSION TITLE
leaf values with plot_tree()

### DISCUSSION THREAD

**Discussion by 14mission** [catboost/catboost#2832] [General]
When I train a classifier, and use plot_tree(), all the leaf node values are fairly small (positive or negative), although the target values for this binary classifier should be 0 or 1.  When the model is actually applied, it produces results that are generally either close to 0 or 1, as intended.  From reading around (https://github.com/catboost/catboost/issues/1389 and https://stackoverflow.com/questions/63407613/catboost-plot-tree-understanding for example), I see that the leaf values printed are RawFormulaVal, and it seems that I should call get_scale_and_bias(), and adjust the leaf node values appropriately.   But get_scale_and_bias() always returns (1.0, 0.0), so that will have no effect.  Perhaps I am doing something wrong and getting initial scale and bias values that are later overridden, or need to get them elsewhere.  Perhaps the fact that I am putting data into pools is an issue.
Here's the essentials of what I'm doing.

`
    print("make train_data_pool")
    train_data_pool = Pool(
        data=train_df[features],
        label=train_df[target],
        cat_features=cat_cols,
    )
...
    catboost_model = CatBoostClassifier(**params)
    catboost_model.fit(train_data_pool, plot=False, verbose=True, eval_set=vali_data_pool)

    Predicted = catboost_model.predict_proba(test_data_pool)[:,1]
...
    if args.gvplot != None:
        print("best_iteration: "+str(catboost_model.best_iteration_))
        print("scale_and_bias: "+str(catboost_model.get_scale_and_bias()))
        catboost_model.plot_tree(catboost_model.best_iteration_,train_data_pool).save(args.gvplot)
`
Looks like custom_metric is set to 'AUC', and loss_function is 'CrossEntropy', if that matters.

**Comment by 14mission** [comment]
read up a bit further, think I understand what's up here
